gluten-core/src/main/scala/io/glutenproject/expression/TernaryExpressionTransformer.scala

```diff
@@ -108,6 +108,55 @@ case class StringLocateTransformer(
   }
 }
 
+case class StringSplitTransformerBase(
+    substraitExprName: String,
+    srcExpr: ExpressionTransformer,
+    regexExpr: ExpressionTransformer,
+    limitExpr: ExpressionTransformer,
+    original: StringSplit)
+  extends ExpressionTransformer {
+
+  override def doTransform(args: java.lang.Object): ExpressionNode = {
+    TernaryExpressionTransformer(substraitExprName, srcExpr, regexExpr, limitExpr, original)
+      .doTransform(args)
+  }
+}
+
+class StringSplitTransformer(
+    substraitExprName: String,
+    srcExpr: ExpressionTransformer,
+    regexExpr: ExpressionTransformer,
+    limitExpr: ExpressionTransformer,
+    original: StringSplit)
+  extends StringSplitTransformerBase(substraitExprName, srcExpr, regexExpr, limitExpr, original)
+  with Logging {
+
+  override def doTransform(args: java.lang.Object): ExpressionNode = {
+    if (!regexExpr.isInstanceOf[LiteralTransformer] || !limitExpr.isInstanceOf[LiteralTransformer]) {
+      throw new UnsupportedOperationException(
+        s"Gluten only supports LiteralTransformer as limit/regex for split function.")
+    }
+
+    val limitNode = limitExpr.doTransform(args)
+    val regexNode = regexExpr.doTransform(args)
+    if (!limitNode.isInstanceOf[LiteralNode] || !regexNode.isInstanceOf[LiteralNode]) {
+      throw new UnsupportedOperationException(
+        "Gluten only supports literal input as limit/regex for split function.")
```

Comment:
Let's use this exception message in the above check for `LiteralTransformer`. It looks more straightforward to users. Thanks!